packages/pwa-kit-create-app/scripts/trim-extensions.test.js

```diff
@@ -57,7 +159,96 @@ expect.extend({
     }
 })
 
-const trimExtensions = require('./trim-extensions')
+const mockFs = () => {
+    fs.statSync.mockImplementation((filePath) => {
+        return {
+            isDirectory: () => {
+                return !filePath.endsWith('.jsx')
+            }
+        }
+    })
+    fs.rmSync.mockImplementation(() => jest.fn())
+    fs.readFileSync.mockImplementation(() => jest.fn())
+    fs.existsSync.mockImplementation((filePath) => {
+        if (
+            filePath.includes('featureAComponent') ||
+            filePath.includes('featureBComponent') ||
+            filePath.includes('featureBPage') ||
+            filePath.includes('store-locator') ||
+            filePath.includes('partial')
+        ) {
+            if (
+                filePath.endsWith('.jsx') ||
+                filePath.endsWith('.tsx') ||
+                filePath.endsWith('.js') ||
+                filePath.endsWith('.ts')
+            ) {
+                return false
+            } else {
+                return true
+            }
+        } else {
+            return true
+        }
+    })
+    fs.unlinkSync.mockReturnValue(true)
+}
+
+describe('trim-extensions without config', () => {
+    beforeEach(() => {
+        jest.resetModules()
+    })
+    it('returns early if no plugins is defined', () => {
+        jest.mock('../assets/plugin-config', () => ({
+            plugins: {}
+        }))
+        const trimExtensions = require('./trim-extensions')
+        const consoleSpy = mockConsole('log')
+        trimExtensions('/mock/dir', {})
+        expect(console.log).toHaveBeenCalledWith('No plugins found, skipping trim')
+        consoleSpy.mockRestore()
+    })
+})
+
+describe('trim-extensions with nested directories', () => {
+    beforeEach(() => {
+        jest.resetAllMocks()
+    })
+    beforeEach(() => {
+        fs.readdirSync.mockImplementation((filePath) => {
+            if (filePath.endsWith('/mock/dir')) {
+                return ['src']
+            } else if (filePath.endsWith('src')) {
+                return ['pages', 'route.jsx']
+            } else if (filePath.endsWith('pages')) {
+                return ['store-locator']
+            } else if (filePath.endsWith('partial')) {
+                return ['modal.jsx']
+            } else if (filePath.endsWith('store-locator')) {
+                return ['index.jsx']
+            } else {
+                return []
+            }
```

Comment:
similar to https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2859/files#r2220474583, it's hard to understand why these special cases are here